File: models/company.js
const mongoose = require('mongoose');

/**
 * إعدادات الثوابت العالمية لـ Seday ERP
 * تم التحديث لتشمل التعديلات الضريبية المطلوبة 2026
 */
const GLOBAL_DEFAULTS = {
    INS_EE_PERCENT: 0.11,
    INS_CO_PERCENT: 0.1875,
    MAX_INS_SALARY: 16700, 
    MIN_INS_SALARY: 2325,
    PERSONAL_EXEMPTION: 20000,
    MEDICAL_EXEMPTION_LIMIT: 10000 // الحد الأقصى السنوي للإعفاء الطبي (10000/12 شهرياً)
};

const companySchema = new mongoose.Schema({
    // اسم الشركة
    name: { 
        type: String, 
        required: [true, 'اسم الشركة مطلوب'], 
        trim: true 
    },
    // إيميل الأدمن (الأساسي للتحقق)
    adminEmail: { 
        type: String, 
        required: [true, 'إيميل الأدمن مطلوب'], 
        unique: true,
        lowercase: true,
        trim: true 
    },
    // الإيميل المستخدم في تسجيل الدخول
    email: {
        type: String,
        required: [true, 'بريد تسجيل الدخول مطلوب'],
        unique: true,
        lowercase: true,
        trim: true
    },
    // كلمة المرور مشفرة
    password: {
        type: String,
        required: [true, 'كلمة المرور مطلوبة']
    },
    
    // الإعدادات المرنة للشركة
    settings: {
        insEmployeePercent: { 
            type: Number, 
            default: GLOBAL_DEFAULTS.INS_EE_PERCENT 
        },
        maxInsSalary: { 
            type: Number, 
            default: GLOBAL_DEFAULTS.MAX_INS_SALARY 
        },
        personalExemption: { 
            type: Number, 
            default: GLOBAL_DEFAULTS.PERSONAL_EXEMPTION 
        },
        // إضافة حد الإعفاء الطبي في الإعدادات لدعم الـ Logic الجديد
        medicalExemptionLimit: {
            type: Number,
            default: GLOBAL_DEFAULTS.MEDICAL_EXEMPTION_LIMIT
        },
        companyLogo: { 
            type: String, 
            default: "" 
        },
        isActive: { 
            type: Boolean, 
            default: true 
        }
    },
    
    createdAt: { type: Date, default: Date.now },
    lastSettingsUpdate: { type: Date, default: Date.now }
});

// تحديث تاريخ التعديل تلقائياً قبل الحفظ
companySchema.pre('save', function(next) {
    this.lastSettingsUpdate = Date.now();
    next();
});

/**
 * التصدير النهائي (Vercel Optimized)
 * نستخدم "Company" كاسم للموديل مع التأكد من عدم تكراره
 */
module.exports = mongoose.models.Company || mongoose.model('Company', companySchema);

File: models/company.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

// الثوابت العالمية لنظام السداد 2026 (قوانين الضرائب المصرية)
// دي القيم الافتراضية اللي السيستم بيفتح عليها لأي شركة جديدة
const (
	InsEmployeePercent    = 0.11    // 11% حصة الموظف في التأمينات
	InsCompanyPercent     = 0.1875  // 18.75% حصة الشركة في التأمينات
	MaxInsSalary          = 16700.0 // الحد الأقصى للتأمينات 2026
	MinInsSalary          = 2325.0  // الحد الأدنى للتأمينات
	PersonalExemption     = 20000.0 // الإعفاء الشخصي السنوي للموظف
	MedicalExemptionLimit = 10000.0 // حد الإعفاء الطبي السنوي
	DefaultWorkingDays    = 30      // عدد أيام الشهر الافتراضية للحساب
)

// Company يمثل هيكل بيانات الشركة في النظام
type Company struct {
	ID                 primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	Name               string             `bson:"name" json:"name"`
	AdminEmail         string             `bson:"adminEmail" json:"adminEmail"`
	Email              string             `bson:"email" json:"email"`
	Password           string             `bson:"password" json:"-"`
	Role               string             `bson:"role" json:"role"` // "admin" أو "company"
	Settings           CompanySettings    `bson:"settings" json:"settings"`
	CreatedAt          time.Time          `bson:"createdAt" json:"createdAt"`
	LastSettingsUpdate time.Time          `bson:"lastSettingsUpdate" json:"lastSettingsUpdate"`
}

// CompanySettings الإعدادات القابلة للتخصيص لكل شركة (نقطة رقم 4)
type CompanySettings struct {
	InsEmployeePercent    float64 `bson:"insEmployeePercent" json:"insEmployeePercent"`
	MaxInsSalary          float64 `bson:"maxInsSalary" json:"maxInsSalary"`
	PersonalExemption     float64 `bson:"personalExemption" json:"personalExemption"`
	MedicalExemptionLimit float64 `bson:"medicalExemptionLimit" json:"medicalExemptionLimit"`
	
	// حقول جديدة لزيادة المرونة بين الشركات
	WorkingDays           int     `bson:"workingDays" json:"workingDays"`           // أيام العمل (30 أو 26 أو 22)
	Currency              string  `bson:"currency" json:"currency"`                 // EGP, USD, etc.
	CompanyLogo           string  `bson:"companyLogo" json:"companyLogo"`
	IsActive              bool    `bson:"isActive" json:"isActive"`
}

// NewCompanySettings دالة لإنشاء إعدادات افتراضية عند تسجيل شركة جديدة
func NewCompanySettings() CompanySettings {
	return CompanySettings{
		InsEmployeePercent:    InsEmployeePercent,
		MaxInsSalary:          MaxInsSalary,
		PersonalExemption:     PersonalExemption,
		MedicalExemptionLimit: MedicalExemptionLimit,
		WorkingDays:           DefaultWorkingDays,
		Currency:              "EGP",
		IsActive:              true,
	}
}